lib/pbench/server/api/resources/query_apis/iteration_samples.py

```diff
@@ -0,0 +1,279 @@
+                                "measurement_type": "latency",
+                               ...
+                            },
+                            "@timestamp_original": "1000",
+                            "@generated-by": "cce1f6d53404b43e5a006c8e6d88e1e0",
+                        },
+                        "sort": [1],
+                    }
+                ],
+            },
+            "aggregations": {
+                "controller": {
+                    "doc_count_error_upper_bound": 0,
+                    "sum_other_doc_count": 0,
+                    "buckets": [{"key": "controller_name", "doc_count": 50}],
+                },
+                "name": {
+                    "doc_count_error_upper_bound": 0,
+                    "sum_other_doc_count": 0,
+                    "buckets": [{"key": "Dataset_name", "doc_count": 50}],
+                },
+                "type": {
+                    "doc_count_error_upper_bound": 0,
+                    "sum_other_doc_count": 0,
+                    "buckets": [
+                        {
+                            "key": "throughput",
+                            "doc_count": 50,
+                            "title": {
+                                "doc_count_error_upper_bound": 0,
+                                "sum_other_doc_count": 0,
+                                "buckets": [
+                                    {
+                                        "key": "Gb_sec",
+                                        "doc_count": 50,
+                                        "uid": {
+                                            "doc_count_error_upper_bound": 0,
+                                            "sum_other_doc_count": 0,
+                                            "buckets": [
+                                                {
+                                                    "key": "client_hostname:127.0.0.1-server_hostname:127.0.0.1-server_port:20010",
+                                                    "doc_count": 25
+                                                },
+                                                {
+                                                    "key": "client_hostname:all-server_hostname:all-server_port:all",
+                                                    "doc_count": 25
+                                                }
+                                            ]
+                                        }
+                                    }
+                                ]
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+        """
+        try:
+            return jsonify(
+                {"hits": es_json["hits"], "aggregations": es_json["aggregations"]}
+            )
+        except KeyError as e:
+            raise PostprocessError(
+                HTTPStatus.BAD_REQUEST,
```

Comment:
Is this really a `BAD_REQUEST` or is it an `INTERNAL_ERROR`?

If the response from the Elasticsearch instance does not have `hits` or `aggregations` then is that the callers fault, or does it indicate some other error on our side?

Reply:
Was fixed.